packages/react/__tests__/PublicAPI-test.js

```diff
@@ -124,6 +124,16 @@ beforeEach(() => {
  * the components that we export and their corresponding API.
  */
 test('Public API should only change with a semver change', () => {
+  // deprecation warnings for fields on objects should be ignored
+  const originalWarn = console.warn;
+  const spy = jest.spyOn(console, 'warn').mockImplementation((message) => {
```

Comment:
I think you can add in a mock for our internal helper here: https://github.com/carbon-design-system/carbon/tree/main/packages/react/src/internal/__mocks__ like we do for warning

Then any test that needs to make sure warning isn't emitted can do `jest.mock('../../internal/deprecateFieldOnObject')`

Reply:
Ah, perfect. That's much cleaner!